### PR TITLE
feat(perf-views) Add more advanced filtering to performance overview 

### PR DESCRIPTION
**Pull request by markstory**
Add discover based filtering to performance overview. This makes it possible to focus this view on subsets of data, and lets users narrow down what they want to look at.

I've retained the bare text search acting as a transaction name filter as it is a ergonomic shortcut.

### CODE REVIEW

File: src/sentry/static/sentry/app/stores/tagStore.tsx [hunk 1]
```diff
@@ -153,7 +153,8 @@ const tagStoreConfig: Reflux.StoreDefinition & TagStoreInterface = {
   },
 
   onLoadTagsSuccess(data) {
-    Object.assign(
+    this.state = Object.assign(
```
Comment (markstory) [inline src/sentry/static/sentry/app/stores/tagStore.tsx:156]: This was causing in-place mutation of the tag list which resulted in the searchbar not having the most current data. This issue has been around a _long_ time, but it was really apparent when I was working on the overview.

File: src/sentry/static/sentry/app/views/performance/data.tsx [hunk 2]
```diff
@@ -38,42 +20,46 @@ export const COLUMN_TITLES = [
   'user misery',
 ];
 
-export function generatePerformanceQuery(location: Location): Readonly<NewQuery> {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/data.tsx:41]: This function was only used in tests, and I didn't think we needed it if we tested the resulting eventview which is more important in my opinion.

File: src/sentry/static/sentry/app/views/performance/transactionSummary/index.tsx [hunk 3]
```diff
@@ -75,6 +76,13 @@ class TransactionSummary extends React.Component<Props, State> {
         this.fetchTotalCount();
       }
     }
+
+    if (
+      !isEqual(prevProps.selection.projects, selection.projects) ||
+      !isEqual(prevProps.selection.datetime, selection.datetime)
+    ) {
+      loadOrganizationTags(api, organization.slug, selection);
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/transactionSummary/index.tsx:84]: This was an oversight in the transaction summary changes. Changing the datetime/project selector should update the tag list.

**Review by wmak — APPROVED** [review]
Minor thing, could you add `query` to `transactionSummaryRouteWithQuery` so clicking a transaction keeps the current query?
Otherwise LGTM :+1:

**Comment by markstory** [comment]
> could you add query to transactionSummaryRouteWithQuery so clicking a transaction keeps the current query?

If we retain the query string what should we do with the bare query part that is an implict transaction name filter? Would you expect that condition to be dropped?

**Comment by wmak** [comment]
I think it would make sense to drop the implicit name filter, since you wouldn't want it to remain if you click the breadcrumb to go back to the homepage

**Pull request merged**